src/Plugin/EntityReferenceSelection/OgSelection.php

```diff
@@ -68,13 +68,13 @@ protected function buildEntityQuery($match = NULL, $match_operator = 'CONTAINS')
     $definition = \Drupal::entityTypeManager()->getDefinition($target_type);
 
     if ($bundle_key = $definition->getKey('bundle')) {
-      $bundles = Og::groupTypeManager()->getAllGroupBundles($target_type);
+      $map = Og::groupTypeManager()->getGroupMap();
```

Comment:
We can replace this with a call to `GroupTypeManager::getGroupsForEntityType()`, this can be passed the `$target_type` parameter directly, so we don't need to retrieve the full map and can save some memory.